lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -1414,6 +1420,53 @@ protected function _validateAndCompleteFieldMapping(array &$mapping)
             throw MappingException::missingFieldName($this->name);
         }
 
+        if (
+            PHP_VERSION_ID >= 70400
+            && isset($this->reflClass)
+            && $this->reflClass->hasProperty($mapping['fieldName'])
+        ) {
+            $property = $this->reflClass->getProperty($mapping['fieldName']);
```

Comment:
the body of the if statement should also be factored out into a private method `$mapping = $this->_validateAndCompleteTypedFieldMapping($mapping);`

Reply:
Done